guiEditor/src/components/propertyTab/propertyTabComponent.tsx

```diff
@@ -156,66 +156,73 @@ export class PropertyTabComponent extends React.Component<IPropertyTabComponentP
-                    const snippet = JSON.parse(xmlHttp.responseText);
-                    const oldId = adt.snippetId;
-                    adt.snippetId = snippet.id;
-                    if (snippet.version && snippet.version != "0") {
-                        adt.snippetId += "#" + snippet.version;
-                    }
-                    this.forceUpdate();
-                    if (navigator.clipboard) {
-                        navigator.clipboard.writeText(adt.snippetId);
-                    }
-
-                    const windowAsAny = window as any;
-
-                    if (windowAsAny.Playground && oldId) {
-                        windowAsAny.Playground.onRequestCodeChangeObservable.notifyObservers({
-                            regex: new RegExp(oldId, "g"),
-                            replace: `parseFromSnippetAsync("${adt.snippetId}`,
-                        });
-                    }
-
-                    this.props.globalState.customSave!.action(adt.snippetId).then(() => {
-                    }).catch((err) => {
+        if (this.props.globalState.customSave) {
+            this.props.globalState.customSave!.action(content).then((resolve) => {
+                adt.snippetId = (resolve as unknown as string);
+                this.forceUpdate();
+                if (navigator.clipboard) {
+                    navigator.clipboard.writeText(adt.snippetId);
+                }
+                alert("GUI saved with ID: " + adt.snippetId + " (please note that the id was also saved to your clipboard)");
+            }).catch((err) => {
+                alert(err);
+            });
+        }
+        else {
+            const xmlHttp = new XMLHttpRequest();
+            xmlHttp.onreadystatechange = () => {
+                if (xmlHttp.readyState == 4) {
+                    if (xmlHttp.status == 200) {
+                        const snippet = JSON.parse(xmlHttp.responseText);
+                        const oldId = adt.snippetId;
+                        adt.snippetId = snippet.id;
+                        if (snippet.version && snippet.version != "0") {
+                            adt.snippetId += "#" + snippet.version;
+                        }
+                        this.forceUpdate();
+                        if (navigator.clipboard) {
+                            navigator.clipboard.writeText(adt.snippetId);
+                        }
+    
+                        const windowAsAny = window as any;
+    
+                        if (windowAsAny.Playground && oldId) {
+                            windowAsAny.Playground.onRequestCodeChangeObservable.notifyObservers({
+                                regex: new RegExp(oldId, "g"),
+                                replace: `parseFromSnippetAsync("${adt.snippetId}`,
+                            });
+                        }
+                        alert("GUI saved with ID: " + adt.snippetId + " (please note that the id was also saved to your clipboard)");

```

Comment:
is it possible to unify both calls? either customSave, or the local save, and deal with error/success the same way (with the same function)?

Reply:
Let me see what I can do. :)